### DISCUSSION TITLE
How to get more transmit power on NRF24 using Radiolib library?

### DISCUSSION THREAD

**Discussion by shooka90** [jgromes/RadioLib#259] [Q&A] [answered]
Hello again, 

I am trying to increase the output power of my NRF24 module, but it's restricted by your library. According to your library documentation the only allowed output power are: -18,-12-6 and 0 dbm. My module (E-Byte ML01DP5) can transfer up to 100 mW which is equal to +20 dBm. I tried to modify your codes, but I got lost and couldn't figure it out.

So, the question is:

How to get more transmit power on NRF24 using Radiolib library?

Datasheet E-Byte ML01DP5:
https://www.ebyte.com/en/pdf-down.aspx?id=1136

**Comment by jgromes** [comment, marked as answer]
The allowed power output values are stated in nRF24 datasheet - these are power settings the PA inside nRF24 can be configured to.

These high-power modules usually have an external PA that can't be controlled by the software, it's just a fixed amplifier. For example, Ebyte E22400M30S conatins an SX1268 (maximum output power of 22 dBm) and an external PA with 11 dB gain. So to get to the advertised "1 W output power" you actually have to set the output power in library to 19 dBm, because the external PA will always amplify that by 11 dB to 30 dBm, or 1 W.

You'll have to contact the manufacturer (Ebyte) to get more information about what sort of PA they used, at least what is the gain. Or if you're willing to sacrifice one module, pop the metal shield and take a look underneath - I usually prefer the second method.

**Reply by shooka90** [reply]
I checked the output rf power with a spectrum analyzer and set the power in the library to 0 dBm. The power was about +18 dBm. So it has an external PA. 
Thank you.
